### ISSUE TITLE
Question about the design details of loss function in UniSplat

### ISSUE THREAD

**Issue by wuyingxiong** [chenshi3/UniSplat#7]
Dear authors,
Thank you for open-sourcing the excellent work UniSplat!

I noticed that Omni-Scene adopts a loss combination of voxel loss + fuse loss, and combines context/target views for loss calculation.
Does UniSplat follow a similar design (voxel loss + fuse loss + context/target view fusion) for its loss function?

> https://github.com/WU-CVGL/Omni-Scene/blob/main/model/omni_gs.py

```
        if self.loss_args.weight_recon > 0:
            # RGB loss for omni-gs
            if self.loss_args.recon_loss_type == "l1":
                rec_loss = torch.abs(rgb_gt - render_pkg_fuse["image"])
            elif self.loss_args.recon_loss_type == "l2":
                rec_loss = (rgb_gt - render_pkg_fuse["image"]) ** 2
            loss = loss + (rec_loss.mean() * self.loss_args.weight_recon)
            set_loss("recon", split, rec_loss.mean(), self.loss_args.weight_recon)
        if self.loss_args.weight_recon_vol > 0 and iter < iter_end - 1000:
            # RGB loss for volume-gs
            if self.loss_args.recon_loss_vol_type == "l1":
                rec_loss_vol = torch.abs(rgb_gt - render_pkg_volume["image"])
            elif self.loss_args.recon_loss_vol_type == "l2":
                rec_loss_vol = (rgb_gt - render_pkg_volume["image"]) ** 2
            elif self.loss_args.recon_loss_vol_type == "l2_mask" or self.loss_args.recon_loss_vol_type == "l2_mask_self":
                rec_loss_vol = (rgb_gt * mask_dptm.unsqueeze(2) - render_pkg_volume["image"] * mask_dptm.unsqueeze(2)) ** 2
            loss = loss + (rec_loss_vol.mean() * self.loss_args.weight_recon_vol)
            set_loss("recon_vol", split, rec_loss_vol.mean(), self.loss_args.weight_recon_vol)
```

**Comment by chenshi3** [comment]
Yes, our loss design shares similarities with Omni-Scene, and the specific formulation is detailed in Equation (11) of our paper. Specifically, we render the Gaussians decoded by the dual-branch under both the input view and the t+1 novel view. We also render the Gaussians from the voxel branch under the input view to compute the loss.

**Comment by wuyingxiong** [comment]
Thanks a lot for your quick reply！ I am currently diving into the loss function design and encountered two key questions while experimenting with the code, hoping to get your insights:
**1. Scope of Voxel Loss Application**

- Does the voxel loss act on both context views and target views, or only on context views?
- Is there any difference in the weight of voxel and fused loss between context and target views during training?

**2. Color Inconsistency in Novel Views of Voxel Gaussians**
In my training experiments, I observed a critical issue: The colors of voxel gaussians are well-aligned and accurate in training views (context/target views used for training). However, when rendering novel views (unseen during training like BEV), the colors of voxel gaussians are completely different (even unrelated) from the training views.

- Have you encountered this color inconsistency of voxel Gaussians in novel views during your  experiencs?
- What are the potential causes for the poor color generalization of voxel Gaussians to novel views?

**Comment by chenshi3** [comment]
In my implementation, the voxel loss is applied only to input views, specifically supervising pixels whose depth projections fall within the voxel representation range, with a weight ratio of roughly 1:5 between voxel loss and fused loss. 

Regarding the color inconsistency in novel views like BEV, feed-forward reconstruction methods generally struggle to generalize under such extreme viewpoint changes. I think that even optimization-based Gaussian Splatting methods would face similar challenges in this setting.

**Comment by wuyingxiong** [comment]
Thanks a lot！Recently，I read the paper and I had a  question about the scale smooth loss mentioned in the paper. What is the role of the scale smooth loss and what is its formula?

**Comment by wuyingxiong** [comment]
Hi author, sorry to bother you again.

I encountered a critical problem during the reproduction of UniSplat: the opacity of Pixel-GS will quickly rush to 0.999 and suppress Voxel-GS, resulting in poor fusion effects. I strongly suspect this issue is related to the loss function design.

I have been waiting for your reply  and I am very eager to get your professional guidance. This issue is very important for my subsequent experiments.

Could you please take a moment to reply when you have time? I would really appreciate it!

**Comment by chenshi3** [comment]
> Hi author, sorry to bother you again.
> 
> I encountered a critical problem during the reproduction of UniSplat: the opacity of Pixel-GS will quickly rush to 0.999 and suppress Voxel-GS, resulting in poor fusion effects. I strongly suspect this issue is related to the loss function design.
> 
> I have been waiting for your reply and I am very eager to get your professional guidance. This issue is very important for my subsequent experiments.
> 
> Could you please take a moment to reply when you have time? I would really appreciate it!

Hi, the train code is released. You can refer the design of loss functoin.
